php-frontend/src/main/java/org/sonar/php/metrics/cpd/CpdVisitor.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.metrics.cpd;
+
+import java.io.File;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.TextRange;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.cpd.NewCpdTokens;
+import org.sonar.php.tree.impl.lexical.InternalSyntaxToken;
+import org.sonar.plugins.php.api.tree.CompilationUnitTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.tree.statement.UseStatementTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+public class CpdVisitor extends PHPVisitorCheck {
+
+  private final SensorContext sensorContext;
+  private InputFile inputFile;
+  private NewCpdTokens cpdTokens;
+
+  public CpdVisitor(SensorContext sensorContext) {
+    this.sensorContext = sensorContext;
+  }
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    File file = context().file();
+    FileSystem fileSystem = sensorContext.fileSystem();
+    inputFile = fileSystem.inputFile(fileSystem.predicates().is(file));
+    cpdTokens = sensorContext.newCpdTokens().onFile(inputFile);
+
+    super.visitCompilationUnit(tree);
+
+    cpdTokens.save();
+  }
+
+  @Override
+  public void visitToken(SyntaxToken token) {
+    if (((InternalSyntaxToken) token).isEOF()) {
+      return;
+    }
+
+    String text = token.text();
+
+//    if (text.startsWith("\"") || text.startsWith("'") || text.startsWith("`")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)